src/cheshire/core.clj

```diff
@@ -24,9 +49,12 @@
          generator (.createGenerator
                     ^JsonFactory (or factory/*json-factory*
                                      factory/json-factory)
-                    ^Writer sw)]
-     (when (:pretty opt-map)
+                    ^Writer sw)
+         print-pretty (:pretty opt-map)]
+     (when (= true print-pretty)
```

Comment:
Regardless I think it's cleaner and allows passing any subclass of PrettyPrinter a user creates